packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -107,17 +108,20 @@ class _BottomAppBarState extends State<BottomAppBar> {
 
   @override
   Widget build(BuildContext context) {
-    final CustomClipper<Path> clipper = widget.shape != null
+    final ThemeData theme = Theme.of(context);
+    final BottomAppBarTheme babTheme = BottomAppBarTheme.of(context);
+    final NotchedShape notchedShape = widget.shape ?? babTheme.shape;
+    final CustomClipper<Path> clipper = notchedShape != null
       ? _BottomAppBarClipper(
         geometry: geometryListenable,
-        shape: widget.shape,
+        shape: notchedShape,
         notchMargin: widget.notchMargin,
       )
       : const ShapeBorderClipper(shape: RoundedRectangleBorder());
     return PhysicalShape(
       clipper: clipper,
-      elevation: widget.elevation,
-      color: widget.color ?? Theme.of(context).bottomAppBarColor,
+      elevation: widget.elevation ?? babTheme.elevation ?? _defaultElevation,
+      color: widget.color ?? babTheme.color ?? theme.bottomAppBarColor,
```

Comment:
I don't believe it's actually necessary because `ThemeData.bottomAppBar` will always be set to a default of White (or grey) [in the `ThemeData` constructor](https://github.com/johnsonmh/flutter/blob/affd6de5cbcbe9f7b9482917df1ef229abade312/packages/flutter/lib/src/material/theme_data.dart#L175). So we never pass null to `PhysicalShape`

Reply:
The documentation says that the default is white. Sounds like the truth is that it's whatever the default value of ThemeData.bottomAppBar is.

Reply:
Ohh you're totally right, updated comment.